Drivetrain

Yoel-Perman-Brilliant opened the pull request.

Yoel-Perman-Brilliant committed:
added documentation

Yoel-Perman-Brilliant committed:
added documentation

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -15,77 +18,122 @@
-  public void simulationPeriodic() {}
+    /**
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
```
you dont have to explain it **EVERY TIME**

tal-sitton replied:
you dont have to explain it **EVERY TIME**

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -15,77 +18,122 @@
+    private final Drivetrain drivetrain = Drivetrain.getInstance();
+
+    /**
+     * constructs an instance of OI - a class for interfacing with the robot.
```
this is not the right description

tal-sitton reviewed (changes requested):
In general, plz remove all the pre-written docs in Robot.java

Yoel-Perman-Brilliant committed:
removed documentation because ya'll hate knowledge, also changed OI's description to hopefully make it a not suck as much

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,148 @@
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
+     * </p>
+     * <p>
```
you opened a paragraph, but.... have you closed it?

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,148 @@
+
+    /**
+     * Constants for a PID controller.
+     * Since they were made using {@code addConstantDouble}, they are constants relative to the code itself, but are
```
repetitive

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,148 @@
+
+    /**
+     * Constants for a feed forward controller More info in {@link FeedForwardController}.
+     * Since they were made using {@code addConstantDouble}, they are constants relative to the code itself, but are
```
repetitive

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,148 @@
+        this.gyro = new ADXRS450_Gyro();
+    }
+
+    public double getLeftDistance() {
```
have you configured distance per pulse?
this will return the pulse, because no distancePerPulse was given

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,148 @@
+        return leftEncoder.getDistance();
+    }
+
+    public double getRightDistance() {
```
ditto

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,148 @@
+        rootNamespace.putNumber("right distance", this::getRightDistance);
+        rootNamespace.putNumber("angle", this::getAngle);
+        rootNamespace.putData("reset encoders", new InstantCommand(this::resetEncoders));
+        rootNamespace.putData("reset gyro", new InstantCommand(this::resetGyro));
```
add `drive forward` and `drive backward` to the shuffleboard.

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Main.java:
```diff
@@ -6,19 +6,10 @@
 
```
no need to touch this class

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+import edu.wpi.first.wpilibj.Joystick;
+
+/**
+ * An object which represents an OI, which handles the active control over the robot from the user's side.
```
`An object which represents an OI` - no kidding.
remove

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * An object which represents an OI, which handles the active control over the robot from the user's side.
+ */
+public class OI /*GEVALD*/{
```
add space after the `/`

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+ * An object which represents an OI, which handles the active control over the robot from the user's side.
+ */
+public class OI /*GEVALD*/{
+    private final Joystick leftJoystick;
```
initialize the joystick here

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+ */
+public class OI /*GEVALD*/{
+    private final Joystick leftJoystick;
+    private final Joystick rightJoystick;
```
initialize the joystick here

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+    private final Joystick rightJoystick;
+
+    public OI() {
+        leftJoystick = new Joystick(0);
```
remove

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+
+    public OI() {
+        leftJoystick = new Joystick(0);
+        rightJoystick = new Joystick(1);
```
remove

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
+     * </p>
+     * <p>
```
you opened a paragraph, but.... have you closed it?

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
+
+    private final Drivetrain drivetrain = Drivetrain.getInstance();
+
+    private final OI oi = new OI();
```
oi should be initialized in `robotInit()`

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+        return leftJoystick.getX();
+    }
+
+    public double getLeftY() {
```
why is this here?

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,32 @@
+        return leftJoystick.getY();
+    }
+
+    public double getRightX() {
```
ditto

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/RobotMap.java:
```diff
@@ -1,13 +1,28 @@
 
+/**
+ * <p>
+ * A class which holds the ports of the devices which are connected to the robot.
```
why is this a paragraph?

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/RobotMap.java:
```diff
@@ -1,13 +1,28 @@
+ * </p>
+ * Each interface represents a different form of connection.
+ *
+ * TODO: remove documentation once there is a readme
```
no, do it now!

Yoel-Perman-Brilliant replied:
it's in update_readme, you just gotta approve it

Yoel-Perman-Brilliant committed:
you had a lot of problems

Yoel-Perman-Brilliant committed:
updated documentation

Yoel-Perman-Brilliant committed:
deleted useless import

Yoel-Perman-Brilliant committed:
deleted useless import

Yoel-Perman-Brilliant committed:
updated doc

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,153 @@
+    private static final RootNamespace corrections = new RootNamespace("corrections");
+
+    /**
+     * Constants which are used for fixing the drivetrain's deviation. More info in {@link BustedMotorControllerGroup}.
```
change `which` to `that`

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,153 @@
+    private final Supplier<Double> kS = rootNamespace.addConstantDouble("kS", 0);
+    private final Supplier<Double> kV = rootNamespace.addConstantDouble("kV", 0);
+    private final Supplier<Double> kA = rootNamespace.addConstantDouble("kA", 0);
+    private final Supplier<Double> kG = rootNamespace.addConstantDouble("kG", 0);
```
why is `kG` here? its a drivetrain

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,153 @@
+    private final Supplier<Double> kP = rootNamespace.addConstantDouble("kP", 0);
+    private final Supplier<Double> kI = rootNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kD = rootNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> TOLERANCE = rootNamespace.addConstantDouble("TOLERANCE", 0);
```
why is my shuffleboard yelling?
(why all caps in shuffleboard)

Yoel-Perman-Brilliant replied:
it's a constant

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,153 @@
+    private final Supplier<Double> kI = rootNamespace.addConstantDouble("kI", 0);
+    private final Supplier<Double> kD = rootNamespace.addConstantDouble("kD", 0);
+    private final Supplier<Double> TOLERANCE = rootNamespace.addConstantDouble("TOLERANCE", 0);
+    private final Supplier<Double> WAIT_TIME = rootNamespace.addConstantDouble("WAIT_TIME", 1);
```
why is my shuffleboard yelling?
(why all caps in shuffleboard)

Yoel-Perman-Brilliant replied:
it's a constant

Yoel-Perman-Brilliant committed:
things

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+public class Drivetrain extends TankDrivetrain {
+
+    /**
+     * <p>
```
convention B-I\b9 - Use both opening and closing tags for paragraph separation, **both inlined**

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+    /**
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
+     * </p>
```
convention B-I\b9 - Use both opening and closing tags for paragraph separation, **both inlined**

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+
+    /**
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
```
does it hold only one value?

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/RobotMap.java:
```diff
@@ -4,10 +4,17 @@
     
     public interface DIO {
-
+        int LEFT_ENCODER_A = 0;
```
Convention A-II/b2, the first line in a class is blank

OfriRosenbaum commented on src/main/java/frc/robot/subsystems/feeder/RobotMap.java:
```diff
@@ -3,7 +3,7 @@
 
     public interface CAN {
-
+        int FEEDER_VICTOR = 0;
```
![image](https://user-images.githubusercontent.com/71877371/155012422-43b67bca-c3d6-4f8b-b87b-c2506cfc4d95.png)

tal-sitton replied:
![image](https://user-images.githubusercontent.com/53299816/155890362-6aa0dbd0-a202-4186-b343-a3d24acdf33d.png)

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
+     */
+    private final RootNamespace robotNamespace = new RootNamespace("robot");
+
+    private final Drivetrain drivetrain = Drivetrain.getInstance();
```
use the `configureDashboard()` in `robotInit()`

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
-  public void simulationPeriodic() {}
+    /**
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
```
does it hold only value?

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
-  @Override
-  public void simulationPeriodic() {}
+    /**
+     * <p>
```
convention B-I\b9 - Use both opening and closing tags for paragraph separation, **both inlined**

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
+    /**
+     * <p>
+     * A namespace is an object which holds value on a {@link NetworkTable}.
+     * </p>
```
convention B-I\b9 - Use both opening and closing tags for paragraph separation, **both inlined**

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,84 @@
+    public void robotInit() {
+        oi = new OI();
+        DriveArcade driveArcade = new DriveArcade(drivetrain, oi::getLeftX, oi::getRightY);
+        /*
```
add a blank line before the javadoc

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,22 @@
+ * Handles the active control over the robot from the user's side.
+ */
+public class OI /*GEVALD*/ {
+    private final Joystick leftJoystick = new Joystick(0);
```
Convention A-II/b2, the first line in a class is blank

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Main.java:
```diff
@@ -5,6 +5,7 @@
 package frc.robot.drivetrains.tankdrivetrain;
 
 import edu.wpi.first.wpilibj.RobotBase;
+import frc.robot.subsystems.climber.Robot;
```
![image](https://user-images.githubusercontent.com/71877371/155013236-7f324dd1-bf91-4945-a238-826d9790781b.png)

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+        return rightEncoder.getDistance();
+    }
+
+    public double getAngle() {
```
this method is simply incorrect

Yoel-Perman-Brilliant replied:
no u

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+        rootNamespace.putData("reset gyro", new InstantCommand(this::resetGyro));
+        rootNamespace.putData("drive forward", new DriveArcade(this, 0.5, 0));
+        rootNamespace.putData("drive backward", new DriveArcade(this, -0.5, 0));
+
```
remove blank line

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+    }
+
+    /**
+     * sets the values which appear on the shuffleboard using a {@link Namespace}.
```
Capital letter at the start of a sentence

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+    private final PIDSettings pidSettings = new PIDSettings(kP, kI, kD, TOLERANCE, WAIT_TIME);
+
+    /**
+     * Constants for a feed forward controller. More info in {@link FeedForwardController}.
```
make it a `@link`, and remove the "More info in..." part

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+            corrections.addConstantDouble("right correction", 1);
+
+    /**
+     * The wheel moves 15.24 * PI (its perimeter) each 360 ticks (in meters).
```
What is distance per pulse?

OfriRosenbaum commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,152 @@
+    public static Drivetrain getInstance() {
+        if (instance == null) {
+            instance = new Drivetrain();
+            instance.configureDashboard();
```
use `configureDashboard()` in `robotInit()`, not here

Yoel-Perman-Brilliant committed:
respectfully, suck it

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,149 @@
+ * This class represents a type of Drivetrain that its left and right sides are controlled independently, allowing it to
+ * move by giving each side a speed value separately.
+ * It can move forward/backward by giving both its sides an equal speed or turn by giving the sides different speeds.
+ * In addition, this drivetrain includes encoders, settings for PID and FeedForward controllers, and communication with
```
and GYRO

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,149 @@
+public class Drivetrain extends TankDrivetrain {
+
+    /**
+     * <p> A namespace is an object which holds values on a {@link NetworkTable}. </p>
```
object `that` holds

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,149 @@
+
+    private final FeedForwardSettings ffSettings = new FeedForwardSettings(kS, kV, kA);
+
+    private static Drivetrain instance;
```
explain that this is singleton

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/OI.java:
```diff
@@ -0,0 +1,23 @@
+import edu.wpi.first.wpilibj.Joystick;
+
+/**
+ * Handles the active control over the robot from the user's side.
```
change

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java:
```diff
@@ -4,88 +4,79 @@
 public class Robot extends TimedRobot {
 
+    /**
+     * <p> A namespace is an object which holds value on a {@link NetworkTable}.</p>
```
ditto

Yoel-Perman-Brilliant committed:
listen I aint gonna go over every single problem you had with the code
also the documentations for getAngle() and getModifiedAngle() include mathematical notations

Yoel-Perman-Brilliant committed:
resolved hidden conversations

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,162 @@
+public class Drivetrain extends TankDrivetrain {
+
+    /**
+     * <p> A namespace is an object that holds values on a {@link NetworkTable}. </p>
```
`in a...`

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,162 @@
+
+    /**
+     * The distance the robot moves every encoder pulse.
+     * The wheel moves 15.24 * PI (its perimeter) each 360 ticks (in meters).
```
change `The` to `This` or `Our`

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,162 @@
+    /**
+     * Sets the values which appear on the shuffleboard using a {@link Namespace}.
+     */
+    public void configureDashboard() {
```
add the option to drive for 2 meters using PID

Yoel-Perman-Brilliant committed:
minor fixes and additions

tal-sitton commented on src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java:
```diff
@@ -0,0 +1,168 @@
+    /**
+     * constants for a PID controller
+     */
+    private final Supplier<Double> kP = rootNamespace.addConstantDouble("kP", 0);
```
change to 1

Yoel-Perman-Brilliant committed:
changed kP to 1 and changed DriveTankWithPID to DriveArcadeWithPID

Yoel-Perman-Brilliant committed:
Merge remote-tracking branch 'origin/dev' into drivetrain

# Conflicts:
#	src/main/java/frc/robot/drivetrains/tankdrivetrain/Robot.java
